core/model/src/main/java/org/eclipse/sw360/antenna/api/exceptions/AntennaInvalidArgumentException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.api.exceptions;
+
+public class AntennaInvalidArgumentException extends AntennaException {
+    public AntennaInvalidArgumentException() {
```

Comment:
I am not a fan of new Exceptions, why is this not simply an IllegalArgumentException. What value does the Antenna prefix bring at this point?

Reply:
Since this an Invalid Argument when it comes to Antenna argument input, an Antenna specific exception should be thrown. 
It is not like they are giving an invalid path, but rather a format that Antenna isn't supporting.

Reply:
From my experience, defining Exceptions is a bad thing to do, it is redundant code and therefore overhead. So the AntennaException is not the best decision there is, in the first place. Especially choosing AntennaException to not be a RuntimeException is from my point of view a real mistake. I know that there is a definition from the Java world about the meaning of Exception and RuntimeException, but the handling of Exception in maintainable code is simply extremely awkward. For example, sometimes you are sure that an exception cannot occur in your code, if this exception is not a runtime exception, you dirty up your code with a catch clause or you have to throw the exception down to the root caller. Both are not really nice solutions. A runtime exception can be simply omitted, the advantage is, that if for some obscure reason your environment changes and the exception can happen, you will face it with an uncaught exception which allows you to investigate, if you silently caught the exception, this you will not face.

Reply:
I see the checked exception as an advantage and that was the reason why I suggested to use that here.

If you prefer unchecked exceptions, we should wirte guidelines and use them.

Reply:
If it was a `IllegalArgumentException` one would have to catch `IllegalArgumentException` for handling it, which is a to broad class of exceptions and hides many programming mistakes and other problems. Thus namespacing the exceptions helps us to only catch exceptions to which we are also responsible for throwing.

Reply:
There are better argumentations against checked exceptions:

- https://phauer.com/2015/checked-exceptions-are-evil/
- https://www.javaworld.com/article/3142626/are-checked-exceptions-good-or-bad.html

But for this PR my proposal is to get rid of the extra exception and use the standard AntennaException for now. I do not see the value of an extra exception.

One point on our last remark @maxhbr. There is no point in catching only the exception coming from us, if in a call an IllegalArgumentException is raised, this means that there is a serious issue in the call stack which has not been handled. Most likely the operation is not executed properly, independent of whether in our code or not. IllegalArgumentException and IllegalStateException have been introduced to explicitely express this. They are used e.g. in Guava.Preconditions to have a standard precondition check in a method. Reinventing the wheel at this point is imho superfluous.

Reply:
> There is no point in catching only the exception coming from us, ...

We know the cause of our specific exception and can handle it by logging a warning. I think that code of the kind
```java
try {
    ...
} catch (IllegalArgumentException e) {
    LOGGER.warn(e.getMessage());
}
```
is bad practice since it hides many more unexpected errors, which we do not want to handle here but not catch at all.